File: Eulerc/Euler10.c
#include<stdio.h>
#include<math.h>
//compile with -lm flag

int isprime(int n){
	if(n < 2){
		return 0;
	}
	if(n == 2){
		return 1;
	}
        int lim = (int)sqrt((double)n);
        if(n % 2 == 0){
                return 0;
        }
        for(int i = 3; i <= lim + 1; i += 2){
                if(n % i == 0){
                        return 0;
                }
        }
        return 1;
}

int main(void){
	long int sum = 2;
	for(int i = 3; i <= 2000001; i++){
		if(isprime(i)){
			sum += i;
		}
	}
	printf("%ld\n", sum);
}

File: Eulergo/Euler10.go
package main

import "fmt"
import "math"

func isPrime(n int) bool {
	if n == 2 {
		return true
	}
	if n < 2 || n % 2 == 0 {
		return false
	}
	f := math.Sqrt(float64(n))
	y := int(f) + 1 
	for i := 3; i <= y ; i += 2 {
		if n % i == 0 {
			return false
		}
	}
	return true
}

func main(){
	sum := 0
	for i := 0; i <= 2000001; i++ {
		if isPrime(i) {
			sum += i
		}
	} 
	fmt.Println(sum)
}

